pkg/controllers/allocation/controller.go

```diff
@@ -105,5 +127,50 @@ func (c *Controller) Reconcile(ctx context.Context, object client.Object) (recon
 			zap.S().Errorf("Continuing after failing to bind, %s", err.Error())
 		}
 	}
-	return reconcile.Result{RequeueAfter: c.Interval()}, nil
+	return reconcile.Result{}, nil
+}
+
+// Watches returns the necessary information to create a watch
+//   a. source: the resource that is being watched
+//   b. eventHandler: which controller objects to be reconciled
+//   c. predicates: which events can be filtered out before processed
+func (c *Controller) Watches(ctx context.Context) (source.Source, handler.EventHandler, builder.WatchesOption) {
+	return &source.Kind{Type: &v1.Pod{}},
+		handler.EnqueueRequestsFromMapFunc(handler.MapFunc(
+			func(obj client.Object) []reconcile.Request {
+				pod := obj.(*v1.Pod)
+				if pod.Spec.NodeName != "" || podutils.IsOwnedByDaemonSet(pod) {
+					return nil
+				}
+				provisioner, err := c.filter.GetProvisionerFor(ctx, pod)
+				if err != nil {
+					zap.S().Errorf("Retrieving provisioner, %s", err.Error())
+					return nil
+				}
+				c.batcher.Add(fmt.Sprintf("%s/%s", provisioner.Name, provisioner.Namespace))
+				requests := []reconcile.Request{{
+					NamespacedName: types.NamespacedName{
+						Name:      provisioner.Name,
+						Namespace: provisioner.Namespace,
+					}}}
+				return requests
+			},
+		)),
+		builder.WithPredicates(
+			predicate.Funcs{
```

Comment:
I wish there were a less verbose way to express this. Perhaps just inlining the functions? If you leave them null, presumably that will default to true?

Reply:
Since I moved the logic to check if we should add to a batch to the mapper, I just went back to processing all events to make it cleaner.